monkey/monkey_island/cc/ui/src/components/report-components/ransomware/BreachSection.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {useEffect, useState} from 'react';
+import IslandHttpClient from '../../IslandHttpClient';
+import NumberedReportSection from './NumberedReportSection';
+import LoadingIcon from '../../ui-components/LoadingIcon';
+import {renderLimitedArray} from '../common/RenderArrays';
+
+function BreachSection() {
+  const [machines, setMachines] = useState(null);
+  let description = 'Ransomware attacks start after machines in the internal network get compromised. ' +
+    'The initial compromise was simulated by running monkeys manually.';
+
+  useEffect(() => {
+    IslandHttpClient.get('/api/exploitations/manual')
+      .then(resp => setMachines(resp.body['manual_exploitations']));
+  }, []);
```

Comment:
Do we get any benefit from having state? Why not:

```suggestion
  let machines = null;
  useEffect(() => {
    IslandHttpClient.get('/api/exploitations/manual')
      .then(resp => machines = resp.body['manual_exploitations']);
  }, []);
```

Reply:
Also, consider moving this to `ReportPage.js` and passing it as a prop to this component. That way this component can have the responsibility of rendering the data, but remain decoupled from the API.

Reply:
Because the section would not re-render. React components represent the state, not local variables.

Reply:
If it has no state, won't it always re-render when the parent component re-renders?

Reply:
![image](https://user-images.githubusercontent.com/19957806/126981857-61bde050-d195-4c4c-81ce-73a41f0445dc.png)
https://felixgerschau.com/react-rerender-components/#when-does-react-re-render